openshiftcontrolplane/v1/types.go

```diff
@@ -296,6 +296,11 @@ type BuildDefaultsConfig struct {
 
 	// resources defines resource requirements to execute the build.
 	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// buildCSIVolumes defines whether the instances of DockerBuildStrategy or SourceBuildStrategy
+	// can employ CSI volumes with any BuildVolume they specify.  Currently, this is only
+	// supported on clusters where the TechPreview feature set is enabled.
+	BuildCSIVolumes bool `json:"buildCSIVolumes,omitempty"`
```

Comment:
this seems odd to be nested under "Defaults" vs being somewhere in the top level BuildControllerConfig object.

Reply:
I tend to agree .... but the input from the description in https://issues.redhat.com/browse/BUILD-281 -

Openshift-controller-manager-operator needs to read the feature gate, pass it on to the build controller.
The build controller has its own configuration "API" - this was a relic of the 3.x master configuration that is not exposed to admins in OCP 4.x: https://github.com/openshift/api/blob/master/openshiftcontrolplane/v1/types.go#L198-L207

@adambkaplan - perhaps a hiccup during planning, and we should instead update https://github.com/openshift/api/blob/master/config/v1/types_build.go#L47-L77 ?

if not, please clarify

Reply:
did you get those links backwards?
https://github.com/openshift/api/blob/85977bee07221f012896dcc53b77f44da0be0c4e/openshiftcontrolplane/v1/types.go#L198-L207 is where i'd expect to put this (as a peer to BuildDefaults, not nested within BuildDefaults).

i'm not sure if this new field warrants a whole new struct within BuildControllerConfig, or if we can just put the bool field directly into that struct....i'm inclined towards the latter, since the struct is already named "BuildControllerConfig"

Reply:
no that is what I was thinking, but I like your take better @bparees 

first things first, and the change that greatly simplifies things on the OCM-O side, is settling on config/v1/types_builds.go instead of openshiftcontrolplane/v1/types.go

hopefully @adambkaplan does not have any gotcha for us when he sees this thread

Reply:
ok, i'm clearly forgetting how we ended up with both:

https://github.com/openshift/api/blob/85977bee07221f012896dcc53b77f44da0be0c4e/config/v1/types_build.go#L19

and

https://github.com/openshift/api/blob/85977bee07221f012896dcc53b77f44da0be0c4e/openshiftcontrolplane/v1/types.go#L198

I suspect you're going to need to change both.  One is presumably the config resource that admins edit/stored in etcd, and the other is used internally to write/hold the config for the controller.

but my point was independent of that and was only that the new field should be a peer to the BuildDefaults field, not a member of its struct.

Reply:
ok @bparees @adambkaplan 

1) field moved up a level
2) analogous fields in both config/v1 and openshiftcontrolplane/v1

PTAL